processor/groupbytraceprocessor/event.go

```diff
@@ -70,25 +92,40 @@ type eventMachine struct {
 	closed       bool
 }
 
-func newEventMachine(logger *zap.Logger, bufferSize int) *eventMachine {
+func newEventMachine(logger *zap.Logger, bufferSize int, numWorkers int, numTraces int) *eventMachine {
 	em := &eventMachine{
 		logger:                    logger,
-		events:                    make(chan event, bufferSize),
+		workers:                   make([]*eventMachineWorker, numWorkers),
 		close:                     make(chan struct{}),
 		shutdownLock:              &sync.RWMutex{},
 		metricsCollectionInterval: time.Second,
 		shutdownTimeout:           10 * time.Second,
 	}
+	for i := range em.workers {
+		em.workers[i] = &eventMachineWorker{
+			machine: em,
+			buffer:  newRingBuffer(numTraces / numWorkers),
```

Comment:
Might be useful to mention in README that numTraces must be at least numWorkers and return error here